src/Multicall3.sol

```diff
@@ -91,22 +92,43 @@ contract Multicall3 {
         (blockNumber, blockHash, returnData) = tryBlockAndAggregate(true, calls);
     }
 
+    event Debug(string x, bytes32 y);
     /// @notice Aggregate calls, ensuring each returns success if required
     /// @param calls An array of Call3 structs
-    /// @return blockNumber The block number where the calls were executed
-    /// @return blockHash The hash of the block where the calls were executed
     /// @return returnData An array of Result structs
-    function aggregate3(Call3[] calldata calls) public payable returns (uint256 blockNumber, bytes32 blockHash, Result[] memory returnData) {
-        blockNumber = block.number;
-        blockHash = blockhash(block.number);
+    function aggregate3(Call3[] calldata calls) public payable returns (Result[] memory returnData) {
         returnData = new Result[](calls.length);
         uint256 length = calls.length;
-        Call3 calldata call;
+        Call3 calldata calli;
         for (uint256 i = 0; i < length;) {
             Result memory result = returnData[i];
-            call = calls[i];
-            (result.success, result.returnData) = call.target.call(call.callData);
-            require(call.allowFailure || result.success, "Multicall3: aggregate3 failed");
+            calli = calls[i];
+            (result.success, result.returnData) = calli.target.call(calli.callData);
+
+            // --- START DEBUG TO VERIFY VALUES ---
+            bytes32 allowFailureTest;
+            bytes32 successTest;
+            bytes32 res;
+            assembly {
+                allowFailureTest := calldataload(add(calli, 32))
+                successTest := mload(result)
+                res := iszero(or(allowFailureTest, successTest))
+            }
+            emit Debug("allowFailureTest", allowFailureTest);
+            emit Debug("successTest     ", successTest);
+            emit Debug("res             ", res);
+            // --- END DEBUG ---
+    
+            assembly {
+                // allowFailure is second element in the array, so load calldata one slot past the calli start
+                let allowFailure := calldataload(add(calli, 32))
+                // success is the first element in the result, so load that directly
+                let success := mload(result)
+                if iszero(or(allowFailure, success)) {
+                    // TODO: FOUNDRY DOESN'T CATCH THIS REVERT?
+                    revert(0, "Multicall3: aggregate3 failed")
```

Comment:
This isnt how assembly reverts work - assembly reverts are `revert(offset, size)`, this evaluates to 0 offset, bytes32("Multicall3: aggregate3 failed") as the size

Reply:
the correct way would be:
```solidity
assembly {
            // set Error(string) signature
            mstore(0x00, 0x08c379a000000000000000000000000000000000000000000000000000000000)
            // set data offset
            mstore(0x04, 0x0000000000000000000000000000000000000000000000000000000000000020)
            // set length
            mstore(0x24, 0x000000000000000000000000000000000000000000000000000000000000001d)
            // set revert string
            mstore(0x44, 0x4d756c746963616c6c333a2061676772656761746533206661696c6564000000)
            revert(0x00, 0x64)
}
```

Reply:
done 🙏